### PR TITLE
Offline dashboard notification error state

### PR DESCRIPTION
**Pull request by szabinst**
refs: MBL-16998
affects: Student
release note: none 
test plan: Start download, turn off Wi-Fi, check notification. 

Notes: 
1. The actual retry logic needs some polish, please focus on the dashboard notification handling for now. 
2. As discussed earlier, the error messages are not yet finalized. 

## Screenshots

https://github.com/instructure/canvas-ios/assets/110813321/5d05da7c-23b3-4323-9fd7-4f8be096a02e

## Checklist

- [x] A11y checked
- [ ] Tested on phone
- [x] Tested on tablet
- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student

[MBL-16998](https://instructure.atlassian.net/browse/MBL-16998)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 92.45% | 92.45% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a93808bffcc68b8dca1b678420eaa26fc2ecdd5e
</p>

File: Core/Core/Dashboard/OfflineSync/ViewModel/DashboardOfflineSyncProgressCardViewModel.swift [hunk 1]
```diff
@@ -35,7 +49,7 @@ class DashboardOfflineSyncProgressCardViewModel: ObservableObject {
 
     /**
      - parameters:
-        - offlineModeInteractor: This is used to determine if the feature flag is turned on. If it's off then
+     - offlineModeInteractor: This is used to determine if the feature flag is turned on. If it's off then
```
Comment (vargaat) [inline Core/Core/Dashboard/OfflineSync/ViewModel/DashboardOfflineSyncProgressCardViewModel.swift:52]: This won't produce the expected documentation:
<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

**Review by vargaat — CHANGES_REQUESTED** [review]
I noticed some strange behavior:

1. Tapping on the failed banner and canceling the sync doesn't make the banner to go away. (It wiggles however on the dashboard like it is about to animate out then coming back.) Tapping it again will show an in-progress sync screen.
2. Restarting the app while the banner is on screen doesn't bring the failed banner back. I would expect to stay it on screen to remind the user that something went wrong during the last sync.
3. Tapping on the banner crashed the app multiple times, although it seems that this is the crash we fixed last week.

Optional improvements:
1. The card animates in/out and the progress updates also animates but there's no fade animation when the card changes from the in progress state to the error state.
2. When using voiceover the card's text is read and it states that it's a button but it's not clear what the button will do. I would make the whole card one a11y element with two actions: "show sync details" and "dismiss". In this case the dismiss button could be hid making the card's handling more straightforward.

**Comment by szabinst** [comment]
@vargaat Fixed your findings, except the crash. I suspect that my upcoming PR will fix the crash.

**Review by vargaat — APPROVED** [review]
QA+1

**Pull request merged**